File: backend/src/main/java/ch/morgias/cookgenda/models/shopping/ShoppingListFood.java
package ch.morgias.cookgenda.models.shopping;

import ch.morgias.cookgenda.models.food.Food;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
public class ShoppingListFood {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id ;
    @ManyToOne
    private ShoppingList shoppingList;
    @ManyToOne(fetch = FetchType.LAZY)
    private Food food;
    private LocalDate planedDate;
    private Double quantity;

    private Boolean checked = false;
}

File: android/app/src/main/java/ch/morgias/cookgenda/android/models/ShoppingListFood.kt
package ch.morgias.cookgenda.android.models

import ch.morgias.cookgenda.android.models.utils.LocalDateSerializer
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDate

@Serializable
data class ShoppingListFood(
    val id: Long,
    val foodId: Long,
    val shoppingListId: Long,
    val name: String,
    val quantity: Double,
    @Serializable(with = LocalDateSerializer::class) @Contextual val planedDate: LocalDate,
    val checked: Boolean,
)